app/addons/fauxton/components.react.jsx

```diff
@@ -84,9 +84,129 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  var Tray = React.createClass({
+
+    getInitialState: function () {
+      return {
+        show: false
+      };
+    },
+
+    toggle: function (done) {
+      if (this.state.show) {
+        this.hide(done);
+      } else {
+        this.show(done);
+      }
+    },
+
+    componentDidMount: function () {
+      $('body').on('click.Tray-' + this.props.trayid + "-" + this._rootNodeID, function (e) {
```

Comment:
I found a better solution for that one - thanks!
BTW: while testing I realized that TestUtils.renderIntoDocument drops our container. So we can get more stable results by just using React.renderComponent